python/paddle/fluid/layers/nn.py

```diff
@@ -5281,3 +5282,74 @@ def crop(x, shape=None, offsets=None, name=None):
         outputs={'Out': out},
         attrs=None if len(attrs) == 0 else attrs)
     return out
+
+
+def rank_loss(label, left, right, name=None):
+    """
+    **Rank loss layer for RankNet**
+
+    RankNet(http://icml.cc/2015/wp-content/uploads/2015/06/icml_ranking.pdf)
+    is a pairwise ranking model with one training sample consisting of a pair
+    of doc A and B, and the label P indicating that A is ranked higher than B
+    or not:
+    
+    P = {0, 1} or {0, 0.5, 1}, where 0.5 means no information about the rank of
+    the input pair.
+    
+    This layer takes three inputs: left (o_i), right (o_j) and label
+    (P_{i,j}), which represent the output score of RankNet for the two docs and
+    the label respectively, and yields the rank loss C_{i,j} using the following
+    equation:
+    
```

Comment:
Rank loss layer takes three inputs: left (o_i), right (o_j) and label (P_{i,j}). The inputs respectively represent RankNet's output scores for documents A and B and the value of label P. The following equation computes rank loss C_{i,j} from the inputs:

Reply:
Done, thanks!